UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -436,9 +446,52 @@ ParseOptions (
 
         PayloadBase->Entry = (EFI_PHYSICAL_ADDRESS)StartAddress;
       }
+    } else if (AsciiStrCmp (NodePtr->Name, "upl-images") == 0) {
+      DEBUG ((DEBUG_INFO, "  Found upl-images node \n"));
+
+      //
+      // Create an empty FvHob for the DXE FV that contains DXE core.
+      //
+      BuildFvHob ((EFI_PHYSICAL_ADDRESS)0, 0);
+
+      DEBUG ((DEBUG_INFO, "UPL FVs found in UPL FDT\n"));
+      for (ISubNode = FdtFirstSubnode (Fdt, SubNode); ISubNode >= 0; ISubNode = FdtNextSubnode (Fdt, ISubNode)) {
+        NodePtr = (FDT_NODE_HEADER *)((CONST CHAR8 *)Fdt + ISubNode + Fdt32ToCpu (((FDT_HEADER *)Fdt)->OffsetDtStruct));
+        DEBUG ((DEBUG_INFO, "\n      SubNode(%08X)  %a", ISubNode, NodePtr->Name));
+
+        // NB: This is not specced!
+        PropertyPtr = FdtGetProperty (Fdt, ISubNode, "name", &TempLen);
+        if (TempLen <= 0) {
+          ASSERT (FALSE);
+          continue;
+        }
+        TempStr = (CHAR8 *)(PropertyPtr->Data);
+
+        PropertyPtr = FdtGetProperty (Fdt, ISubNode, "reg", &TempLen);
+        if (TempLen <= 0) {
+          ASSERT (FALSE);
+          continue;
+        }
+        Data64        = (UINT64 *)(PropertyPtr->Data);
+        StartAddress  = Fdt64ToCpu (ReadUnaligned64 (Data64));
+        NumberOfBytes = Fdt32ToCpu (*(Data64 + 1));

```

Comment:
Data64 is 64 bit , Argument of Fdt32ToCpu is 32 bit.
this will cause warning C4244.